### PR TITLE
UV/5yr separated dynamic gsplot logic from static styles

### PR DESCRIPTION
**Pull request by thongsav-usgs**

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -4,7 +4,6 @@ export(correctionsataglance)
 export(createDataRows)
```
Comment (mhines-usgs) [inline NAMESPACE]: prob don't need this exported either? (from extremes-data)

File: R/fiveyeargwsum-render.R [hunk 2]
```diff
@@ -74,3 +74,32 @@ createfiveyeargwsumPlot <- function(data){
...
+  args <- list(...)
+  
+  styles <- switch(names(reportObject), 
+      stat = list(
+          lines = append(list(x=x, y=y, legend.name=legend.name), styles$stat_lines)
+          ),
+      est_stat = list(
+          lines = append(list(x=x, y=y, legend.name=legend.name), styles$est_stat_lines)
+          ),
+      max_iv = list(
```
Comment (lindsayplatt) [inline R/fiveyeargwsum-render.R:93]: maybe add a comment here as to why legend.name can't be included in the reportObject? Just so our future selves know

File: R/fiveyeargwsum-styles.R [hunk 3]
```diff
@@ -1,17 +1,11 @@
...
-    styles <- switch(names(data), 
-                     stat = list(lines = list(x=x, y=y, type="o", col="black", pch=20, cex=0.5, lwd=0.8, legend.name=legend.name)),
-                     est_stat = list(lines = list(x=x, y=y, col="red", lty=2, legend.name=legend.name)),
-                     max_iv = list(points = list(x=x, y=y, col="red", pch=8, cex=2, legend.name=paste(args$maxLabel, info$type, ":", y))),
-                     min_iv = list(points = list(x=x, y=y, col="blue", pch=8, cex=2, legend.name=paste(args$minLabel, info$type, ":", y))),
-                     gw_level = list(points = list(x=x,y=y, pch = 8, bg = 'orange', col = 'orange', cex = 1.2, lwd=1, legend.name="Measured Water Level (GWSI)"))
-                     )
-    
-    return(styles)
+getFiveyearStyle <- function() {
```
Comment (lindsayplatt) [inline R/fiveyeargwsum-styles.R:1]: I really like this idea of separating them out :+1:

File: R/fiveyeargwsum-styles.R [hunk 4]
```diff
@@ -1,17 +1,11 @@
...
+  styles <- list(
+      stat_lines = list(type="o", col="black", pch=20, cex=0.5, lwd=0.8),
+      est_stat_lines = list(col="red", lty=2),
+      max_iv_points = list(col="red", pch=8, cex=2),
+      min_iv_points = list(col="blue", pch=8, cex=2),
+      gw_level_points = list(pch = 8, bg = 'orange', col = 'orange', cex = 1.2, lwd=1, legend.name="Measured Water Level (GWSI)")
```
Comment (lindsayplatt) [inline R/fiveyeargwsum-styles.R:7]: should `legend.name` go in the switch statement instead? I know it's hardcoded, but it is more closely tied to the data than plotting styles.

File: R/uvhydrograph-render.R [hunk 5]
```diff
@@ -290,26 +288,17 @@ createSecondaryPlot <- function(data, month, useDownsampled=FALSE){
 
 #' Add UV hydrograph objects to a plot object.
 #' @param object A gsplot, plot object.
-#' @param data Time series data to add to hydrograph.
-#' @param info Meta-data about time series data in "data" parameter.
-#' @param plotName Plot name string. Presently an element of
-#'                 domain {primary,secondary}.
-#' @param dataSides Sides data frame. See getUvStyle().
-#' @param dataLimits Limits data frame. See getUvStyle().
+#' @param list of gsplot calls to make
```
Comment (lindsayplatt) [inline R/uvhydrograph-render.R]: this needs to describe the arguments in the form `@param [arg name] [arg description]` . I think you are missing the argument name here.

File: R/uvhydrograph-render.R [hunk 6]
```diff
@@ -363,3 +352,180 @@ YEndpoint <- function (corr.value.sequence, uncorr.value.sequence) {
   
   return(y.endpoint)
 }
+
+
+#' TODO
```
Comment (lindsayplatt) [inline R/uvhydrograph-render.R]: I think TODO is referring to this, but you need all of the function documentation

File: R/uvhydrograph-render.R [hunk 7]
```diff
@@ -363,3 +352,180 @@ YEndpoint <- function (corr.value.sequence, uncorr.value.sequence) {
...
+  corrArrowPositions <- getCorrectionArrowPositions(x, y, correctionLabels)
+  corrAblinePositions <- getCorrectionAbLines(reportObject)
+  
+  primary_lbl <- info$primary_lbl
+  reference_lbl <- info$reference_lbl
+  comp_lbl <- paste("Comparison", info$comp_UV_TS_lbl, "@", info$comp_UV_lbl)
+  
+  compAxes <- TRUE
+  compAnnotations <- TRUE
+  compLabel <- primary_lbl
```
Comment (lindsayplatt) [inline R/uvhydrograph-render.R:370]: what is the difference between comp_lbl and compLabel? Can we have some comments specifying?

File: R/uvhydrograph-render.R [hunk 8]
```diff
@@ -363,3 +352,180 @@ YEndpoint <- function (corr.value.sequence, uncorr.value.sequence) {
...
+  
+  if(dataSides$comparison == 6){
```
Comment (lindsayplatt) [inline R/uvhydrograph-render.R:372]: can you add some comments about what's going on here?

File: R/uvhydrograph-render.R [hunk 9]
```diff
@@ -363,3 +352,180 @@ YEndpoint <- function (corr.value.sequence, uncorr.value.sequence) {
...
+          ),
+      csg_readings = list(
+          points=append(list(x=x, y=y), styles$csg_readings_points), 
+          error_bar=append(list(x=x, y=y, y.low=reportObject$csg_readings$uncertainty, y.high=reportObject$csg_readings$uncertainty), styles$csg_readings_error_bars)
+          ),
+      hwm_readings = list(
+          points=append(list(x=x, y=y), styles$hwm_readings_points), 
+          error_bar=append(list(x=x, y=y, y.low=reportObject$hwm_readings$uncertainty, y.high=reportObject$hwm_readings$uncertainty), styles$hwm_readings_error_bars)
+          ),
+      appr_approved_dv = list(
```
Comment (lindsayplatt) [inline R/uvhydrograph-render.R:426]: why is pch variable and not static?

File: R/uvhydrograph-render.R [hunk 10]
```diff
@@ -363,3 +352,180 @@ YEndpoint <- function (corr.value.sequence, uncorr.value.sequence) {
...
+          ),
+      appr_working_dv = list(
+          points = append(list(x=x, y=y, pch=reportObject[[1]]$point_type, legend.name=legend.name), styles$appr_working_dv_points)
+          )
+  )
+  
+  return(plotConfig)
+}
+
+#' TODO
```
Comment (lindsayplatt) [inline R/uvhydrograph-render.R]: again, assuming TODO = add function documentation :)

File: R/uvhydrograph-render.R [hunk 11]
```diff
@@ -363,3 +352,180 @@ YEndpoint <- function (corr.value.sequence, uncorr.value.sequence) {
...
+getCorrectionArrowPositions <- function(time, value, correctionLabels) {
+  x <- time
+  y <- value
+  
+  corrArrowPositions <- list()
+  
+  #Make the correction label lines connect to the outside of the bounding box and not to the center of the label
+  if(!isEmptyOrBlank(correctionLabels)){
+    # work around irrelevant warnings from devtools::check()
```
Comment (lindsayplatt) [inline R/uvhydrograph-render.R:508]: you can also use `select_()` instead of `select()` to get around those

File: R/uvhydrograph-render.R [hunk 12]
```diff
@@ -363,3 +352,180 @@ YEndpoint <- function (corr.value.sequence, uncorr.value.sequence) {
...
+  
+  return(plotConfig)
+}
+
+#' TODO
+getCorrectionArrowPositions <- function(time, value, correctionLabels) {
+  x <- time
+  y <- value
+  
+  corrArrowPositions <- list()
```
Comment (lindsayplatt) [inline R/uvhydrograph-render.R:504]: I would remove this and put it in an `else` statement after the L513 stuff. Putting it here makes it look like you need to initialize the variable before being able to do anything else when I think it just needs to return an empty list if `correctionLabels` are empty.

File: R/uvhydrograph-render.R [hunk 13]
```diff
@@ -363,3 +352,180 @@ YEndpoint <- function (corr.value.sequence, uncorr.value.sequence) {
...
+        mutate(x = ifelse(x > xorigin, x - 60 * 60 * 2.85 * correctionLabels$r, x + 60 * 60 * 2.85 * correctionLabels$r)) %>% 
+        as.list()
+  }
+  
+  return(corrArrowPositions)
+}
+
+#' TODO
+getCorrectionAbLines <- function(reportObject) {
+  corrAblinePositions <- list()
```
Comment (lindsayplatt) [inline R/uvhydrograph-render.R:525]: same here - put this in an `else` statement below L527

File: R/uvhydrograph-styles.R [hunk 14]
```diff
@@ -1,102 +1,55 @@
...
-                                    text=list(x=correctionLabels$x, y=correctionLabels$y, labels=correctionLabels$label, srt=0, cex=0.6, side=7, axes=FALSE, pos=1, offset = -0.12, col=rgb(0,0,255,240,maxColorValue=255))),
-
-                gage_height = list(points=list(x=x, y=y, pch=21, bg='black', col='black', cex=.8, lwd=1, legend.name="Measurement gage height"),
-                                   callouts=list(x=x, y=y, labels=data$gage_height$n)),
-                gw_level = list(points = list(x=x,y=y, pch = 8, bg = 'orange', col = 'orange', cex = 1.2, lwd=1, legend.name="Measured Water Level (GWSI)")), 
-                meas_shift = list(points=list(x=x, y=y, pch=21, bg='green', col='green', cex=1, lwd=1, side=4, legend.name="Effective shift and error limits"),
-                                  error_bar=list(x=x, y=y, y.low=(y-data$meas_shift$minShift), y.high=(data$meas_shift$maxShift-y), col='green', lwd=.7, side=4)))
-  } 
+getUvStyles <- function() {
```
Comment (lindsayplatt) [inline R/uvhydrograph-styles.R:4]: I really like this - it's much cleaner!

**Review by lindsayplatt — DISMISSED** [review]
A couple of high-level thoughts:
1. Can you finish documentation before this is merged?
2. Don't forget that we are trying to move away from using $ to using [[ ]] where possible.

File: R/uvhydrograph-render.R (hunk 8 above)
Comment (zmoore-usgs) [reply R/uvhydrograph-render.R:372]: I wrote this part back when adding the third Y-Axis. This is the logic for specifying how to label each side. If any TS are plotted on the same side they should have the same label and not plot their own axis marks, but if they're on different sides they should use different labels and show their axis marks. So this is checking what side each TS is being plotted on and applying the proper label. I can add some comments to clarify the logic in a separate PR after this is merged.

File: R/uvhydrograph-render.R (hunk 7 above)
Comment (zmoore-usgs) [reply R/uvhydrograph-render.R:370]: See my comment below.

**Review by zmoore-usgs — COMMENTED** [review]
I don't have any additional feedback outside of what Lindsay posted. I think this pattern makes sense, so I'll implement for DV Hydro as well. Once this is merged I'll also add comments to my axis labeling logic.

**Comment by thongsav-usgs** [comment]
@lindsaycarr will start moving from [[]] notation, which is now possible since dropping the downsampling stuff. I am intentionally saving documentation for last as the methods are still changing quite a bit. Will incorporate the low hanging fruit and get this merged sooner rather than later because some there is some utils-read changes pending that we should get merged for SRS/SVP work being done by @kmschoep-usgs.

File: R/uvhydrograph-render.R (hunk 12 above)
Comment (thongsav-usgs) [reply R/uvhydrograph-render.R:504]: It's actually considered good practice to always initialize variables to their desired default values. Reasoning being if the logic which comes after does not hit all edge cases you still get the default (eg: if there are nested if/else statements where someone leaves out an edge case, we still get the default value).

File: R/fiveyeargwsum-styles.R (hunk 4 above)
Comment (thongsav-usgs) [reply R/fiveyeargwsum-styles.R:7]: I was thinking of it as more styling than data. This value has changed a couple of times and it's been based on user preference.

**Pull request merged**